lib/Sema/CSApply.cpp

```diff
@@ -1421,30 +1423,47 @@ namespace {
         base = coerceImplicitlyUnwrappedOptionalToValue(base, objTy, locator);
         baseTy = cs.getType(base);
       }
-
-      // Figure out the index and result types.
-      auto subscriptTy = simplifyType(selected->openedType);
-      auto subscriptFnTy = subscriptTy->castTo<AnyFunctionType>();
-      auto resultTy = subscriptFnTy->getResult();
-
+      
+      
+      bool isDynamicMemberLookup =
+        choice.getKind() == OverloadChoiceKind::DynamicMemberLookup;
+      
+      auto locatorKind =
+        isDynamicMemberLookup ? ConstraintLocator::Member
+                              : ConstraintLocator::SubscriptMember;
       // If we opened up an existential when performing the subscript, open
       // the base accordingly.
-      auto knownOpened = solution.OpenedExistentialTypes.find(
+      auto knownOpened =
+        solution.OpenedExistentialTypes.find(
                            getConstraintSystem().getConstraintLocator(
-                             locator.withPathElement(
-                               ConstraintLocator::SubscriptMember)));
+                             locator.withPathElement(locatorKind)));
       if (knownOpened != solution.OpenedExistentialTypes.end()) {
         base = openExistentialReference(base, knownOpened->second, subscript);
         baseTy = knownOpened->second;
       }
 
-      // Coerce the index argument.
-      index = coerceCallArguments(index, subscriptFnTy, nullptr,
-                                  argLabels, hasTrailingClosure,
-                                  locator.withPathElement(
-                                    ConstraintLocator::SubscriptIndex));
-      if (!index)
-        return nullptr;
+      // Figure out the index and result types.
+      Type resultTy;
+      if (!isDynamicMemberLookup) {
+        auto subscriptTy = simplifyType(selected->openedType);
+        AnyFunctionType *subscriptFnTy = subscriptTy->castTo<AnyFunctionType>();
```

Comment:
It will actually always be a FunctionType. GenericFunctionType cannot appear here

Reply:
Fixed, thanks!